### PR TITLE
Update Woodwork to 0.0.7

### PR DESCRIPTION
**Pull request by angela97lin**
Latest Woodwork version adds useful functionality, and addresses some issues (https://github.com/alteryx/woodwork/issues/455) so I think it's a worthwhile upgrade 😁

### CODE REVIEW

File: evalml/tests/dependency_update_check/latest_dependency_versions.txt [hunk 1]
```diff
@@ -18,5 +18,5 @@ scikit-optimize==0.8.1
 scipy==1.5.4
 shap==0.37.0
 texttable==1.6.3
-woodwork==0.0.6
+woodwork==0.0.7
```
Comment (angela97lin) [inline evalml/tests/dependency_update_check/latest_dependency_versions.txt:21]: Odd, our dependency checker should have caught this?

File: evalml/utils/gen_utils.py [hunk 2]
```diff
@@ -294,15 +294,11 @@ def _convert_to_woodwork_structure(data):
     # Convert numpy data structures to pandas data structures
     if isinstance(data, list):
         ww_data = np.array(data)
-
-    if isinstance(ww_data, pd.api.extensions.ExtensionArray) or (isinstance(ww_data, np.ndarray) and len(ww_data.shape) == 1):
```
Comment (angela97lin) [inline evalml/utils/gen_utils.py:298]: Yay, Woodwork update means we can convert directly from numpy now, so no need for this extra logic to convert to pandas first!

**Review by dsherry — APPROVED** [review]
Very cool!!

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by jeremyliweishih — APPROVED** [review]
nice!

**Pull request merged**